tests/verification/integration_test.py

```diff
@@ -0,0 +1,136 @@
+from pathlib import Path
+
+import pytest
+
+from rflx.identifier import ID
+from tests import utils
+from tests.const import MAIN
+from tests.data.fixtures.integration import (
+    DEFINITE_MESSAGE_WITH_BUILTIN_TYPE_SPEC,
+    DEFINITE_PARAMETERIZED_MESSAGE_SPEC,
+    PARAMETERIZED_MESSAGE_SPEC,
+)
+from tests.utils import FEATURES, assert_provable_code, create_complement, create_model, get_config
+
+
+@pytest.mark.parametrize("feature", [f.name for f in FEATURES])
+def test_provability(feature: str, tmp_path: Path) -> None:
+    config = get_config(feature)
+    if config.prove is None:
+        pytest.skip()
+    model, integration = create_model(feature)
+    units = []
+    if model.sessions:
+        assert len(model.sessions) == 1
+        assert model.sessions[0].identifier == ID("Test::Session")
+        units = ["main", "lib", "rflx-test-session"]
+        create_complement(config, feature, tmp_path)
+    assert_provable_code(model, integration, tmp_path, main=MAIN, units=[*units, *config.prove])
+
+
+def test_definite_message_with_builtin_type_provability(tmp_path: Path) -> None:
+    utils.assert_provable_code_string(
+        DEFINITE_MESSAGE_WITH_BUILTIN_TYPE_SPEC, tmp_path, units=["rflx-test-message"]
+    )
+
+
+def test_parameterized_message_provability(tmp_path: Path) -> None:
+    utils.assert_provable_code_string(
+        PARAMETERIZED_MESSAGE_SPEC, tmp_path, units=["rflx-test-message"]
+    )
```

Comment:
These tests are originally from `tests/integration/specification_model_generator_test.py`. While the name was kept for the compilation tests (`tests/compilation/specification_model_generator_test.py`), for the verification tests the name was changed to `integration_test.py`. I would prefer to have the naming consistent. How about using `integration_test.py` in both cases? It is shorter than `specification_model_generator_test.py`.

Reply:
Fixed.